### PR TITLE
pmd:UseProperClassLoader - Use Proper Class Loader

### PR DESCRIPTION
**Pull request by m-ezzat**
This pull request is focused on resolving occurrences of Sonar rule pmd:UseProperClassLoader - Use Proper Class Loader

You can find more information about the issue here: https://dev.eclipse.org/sonar/coding_rules#q=pmd:UseProperClassLoader

Please let me know if you have any questions.

M-Ezzat

### CODE REVIEW

**Comment by m-ezzat** [comment]
I am not sure whether you do not agree with these changes or not. Please close this PR if not.

**Comment by baev** [comment]
@m-ezzat Hi! Sorry for the delay, and thanks for the contribution!

Could you please explain in more details what is wrong with `Class getClassLoader()` in J2EE? As far as I know you can change `Thread.getContextClassLoader()` for some reason and that code might not work as expected as well.

**Comment by m-ezzat** [comment]
Hi @baev I think you will find the answer here 
http://stackoverflow.com/questions/34787419/pmd-rule-use-proper-class-loader-explaination

File: allure-bundle/src/main/java/ru/yandex/qatools/allure/AllureMain.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public static void main(String[] args) throws IOException {
      * @throws IOException if any occurs.
      */
     private static void unpackFace(File outputDirectory) throws IOException {
-        ClassLoader loader = AllureMain.class.getClassLoader();
```
Comment (baev) [inline allure-bundle/src/main/java/ru/yandex/qatools/allure/AllureMain.java]: In that case I'm definitely need to use `AllureMain.class.getClassLoader()`. The resource that I need to unpack is located in `allure-bundle.jar`, and available from ClassLoader that loaded `AllureMain`

**Comment by m-ezzat** [comment]
Ok, it's reverted.

**Comment by baev** [comment]
@m-ezzat thanks!

**Pull request merged**